src/cloud_what/providers/azure.py

```diff
@@ -146,6 +150,16 @@ def get_api_versions(self) -> Union[List[str], None]:
                 api_versions = api_versions_dict['apiVersions']
         return api_versions
 
+    def _has_azure_linux_agent(self):
+        result = False
+        try:
+            with open("/var/log/waagent.log") as f:
```

Comment:
I would like to know why did you decide to check for this file? Why don't you check for presence of `/usr/sbin/waagent`? Or even better why don't you use following construct:

```python
from shutil import which
return which('waagent') is not None
```

Reply:
Sure. This would be possible, too. There is no reason. Its just that I saw this log file.